<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="ns" default=""/>
  <arg name="calibration_service_name" default="calibrate_radar_lidar"/>
  <arg name="rviz" default="true"/>
  <let name="rviz_profile" value="$(find-pkg-share marker_radar_lidar_calibrator)/rviz/default.rviz"/>
  <arg name="radar_parallel_frame" default="front_unit_base_link"/>
  <arg name="input_lidar_pointcloud" default="/sensing/lidar/front_lower/pointcloud_raw"/>
  <arg name="input_radar_objects" default="/sensing/radar/front_center/objects_raw"/>

  <arg name="use_lidar_initial_crop_box_filter" default="false"/>
  <arg name="lidar_initial_crop_box_min_x" default="-50.0"/>
  <arg name="lidar_initial_crop_box_min_y" default="-50.0"/>
  <arg name="lidar_initial_crop_box_min_z" default="-50.0"/>

  <arg name="lidar_initial_crop_box_max_x" default="50.0"/>
  <arg name="lidar_initial_crop_box_max_y" default="50.0"/>
  <arg name="lidar_initial_crop_box_max_z" default="50.0"/>

  <arg name="use_radar_initial_crop_box_filter" default="false"/>
  <arg name="radar_initial_crop_box_min_x" default="-50.0"/>
  <arg name="radar_initial_crop_box_min_y" default="-50.0"/>
  <arg name="radar_initial_crop_box_min_z" default="-50.0"/>

  <arg name="radar_initial_crop_box_max_x" default="50.0"/>
  <arg name="radar_initial_crop_box_max_y" default="50.0"/>
  <arg name="radar_initial_crop_box_max_z" default="50.0"/>

  <group>
    <push-ros-namespace namespace="$(var ns)"/>
    <node pkg="marker_radar_lidar_calibrator" exec="marker_radar_lidar_calibrator" name="marker_radar_lidar_calibrator" output="screen">
      <param name="radar_parallel_frame" value="$(var radar_parallel_frame)"/>

      <remap from="extrinsic_calibration" to="$(var calibration_service_name)"/>
      <remap from="input_lidar_pointcloud" to="$(var input_lidar_pointcloud)"/>
      <remap from="input_radar_objects" to="$(var input_radar_objects)"/>

      <param name="use_lidar_initial_crop_box_filter" value="$(var use_lidar_initial_crop_box_filter)"/>
      <param name="lidar_initial_crop_box_min_x" value="$(var lidar_initial_crop_box_min_x)"/>
      <param name="lidar_initial_crop_box_min_y" value="$(var lidar_initial_crop_box_min_y)"/>
      <param name="lidar_initial_crop_box_min_z" value="$(var lidar_initial_crop_box_min_z)"/>

      <param name="lidar_initial_crop_box_max_x" value="$(var lidar_initial_crop_box_max_x)"/>
      <param name="lidar_initial_crop_box_max_y" value="$(var lidar_initial_crop_box_max_y)"/>
      <param name="lidar_initial_crop_box_max_z" value="$(var lidar_initial_crop_box_max_z)"/>

      <param name="use_radar_initial_crop_box_filter" value="$(var use_radar_initial_crop_box_filter)"/>
      <param name="radar_initial_crop_box_min_x" value="$(var radar_initial_crop_box_min_x)"/>
      <param name="radar_initial_crop_box_min_y" value="$(var radar_initial_crop_box_min_y)"/>
      <param name="radar_initial_crop_box_min_z" value="$(var radar_initial_crop_box_min_z)"/>

      <param name="radar_initial_crop_box_max_x" value="$(var radar_initial_crop_box_max_x)"/>
      <param name="radar_initial_crop_box_max_y" value="$(var radar_initial_crop_box_max_y)"/>
      <param name="radar_initial_crop_box_max_z" value="$(var radar_initial_crop_box_max_z)"/>
    </node>

    <node pkg="marker_radar_lidar_calibrator" exec="calibrator_ui_node.py" name="calibrator_ui" output="screen"/>
    <node pkg="marker_radar_lidar_calibrator" exec="metrics_plotter_node.py" name="metrics_plotter_node" output="screen"/>

    <node pkg="rviz2" exec="rviz2" name="rviz2" output="screen" args="-d $(var rviz_profile)" if="$(var rviz)">
      <remap from="/pointcloud_topic" to="$(var input_lidar_pointcloud)"/>
    </node>
  </group>
</launch>
